robot-server/robot_server/persistence/__init__.py

```diff
@@ -43,6 +56,10 @@ async def get_persistence_directory(
             )
         else:
             persistence_dir = setting
+            # Reset DB only if is not temporary dir
+            if Path(persistence_dir / _TO_BE_DELETED_ON_REBOOT).exists():
```

Comment:
This doesn't actually matter, but to consistently use async I/O in this function, we can do this:

```suggestion
            if await AsyncPath(persistence_dir / _TO_BE_DELETED_ON_REBOOT).exists():
```